cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -199,15 +209,27 @@ def setup_non_leader_authentication():
     basic_auth = '/srv/kubernetes/basic_auth.csv'
     known_tokens = '/srv/kubernetes/known_tokens.csv'
 
+    hookenv.status_set('maintenance', 'Rendering authentication templates.')
+
+    # Set an array for looping logic
+    keys = [service_key, basic_auth, known_tokens]
+    get_keys_from_leader(keys)
+
+    api_opts.add('--basic-auth-file', basic_auth)
+    api_opts.add('--token-auth-file', known_tokens)
+    api_opts.add('--service-cluster-ip-range', service_cidr())
+    api_opts.add('--service-account-key-file', service_key)
+    controller_opts.add('--service-account-private-key-file', service_key)
+
+    set_state('authentication.setup')
+
+
+def get_keys_from_leader(keys):
     # This races with other codepaths, and seems to require being created first
     # This block may be extracted later, but for now seems to work as intended
     os.makedirs('/etc/kubernetes', exist_ok=True)
     os.makedirs('/srv/kubernetes', exist_ok=True)
 
-    hookenv.status_set('maintenance', 'Rendering authentication templates.')
-
-    # Set an array for looping logic
-    keys = [service_key, basic_auth, known_tokens]
     for k in keys:
         # If the path does not exist, assume we need it
         if not os.path.exists(k):
```

Comment:
If the path doesn't exist and the contents are none this new method now returns, which is different behavior than a return in the original method. Is this intended?